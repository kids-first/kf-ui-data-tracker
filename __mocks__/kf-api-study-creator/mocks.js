import {
  ALL_STUDIES,
  GET_STUDY_BY_ID,
  MY_PROFILE,
  ALL_EVENTS,
  ALL_USERS,
  GET_PROJECTS,
  STATUS,
} from '../../src/state/queries';
import {
  CREATE_PROJECT,
  SYNC_PROJECTS,
  CREATE_STUDY,
  UPDATE_STUDY,
} from '../../src/state/mutations';
import {GET_FILE_BY_ID} from '../../src/documents/queries';
import {
  DELETE_FILE,
  UPDATE_FILE,
  UPDATE_VERSION,
  CREATE_VERSION,
  FILE_DOWNLOAD_URL,
} from '../../src/documents/mutations';
import {GET_DEV_TOKENS} from '../../src/admin/queries';
import {DELETE_DEV_TOKEN, CREATE_DEV_TOKEN} from '../../src/admin/mutations';
import allStudies from './responses/allStudies';
import allStudies_empty from './responses/allStudies_empty';
import studyByKfId from './responses/studyByKfId.json';
import studyByKfId_refetch from './responses/studyByKfId_refetch.json';
import studyByKfId_event_err from './responses/studyByKfId_event_err.json';
import studyByKfId_no_projects from './responses/studyByKfId_no_projects.json';
import studyByKfId_empty from './responses/studyByKfId_empty.json';
import deleteFile from './responses/deleteFile.json';
import fileByKfId from './responses/fileByKfId.json';
import fileByKfId_empty from './responses/fileByKfId_empty.json';
import myProfile from './responses/myProfile.json';
import updateFile from './responses/updateFile.json';
import updateVersion from './responses/updateVersion.json';
import createVersion from './responses/createVersion.json';
import createStudy from './responses/createStudy.json';
import updateStudy from './responses/updateStudy.json';
import devTokens from './responses/devTokens.json';
import createDevToken from './responses/createDevToken.json';
import allEvents from './responses/allEvents.json';
import allEvents_refetch from './responses/allEvents_refetch.json';
import allEvents_20 from './responses/allEvents_20.json';
import allEvents_40 from './responses/allEvents_40.json';
import allUsers from './responses/allUsers.json';
import allProjects from './responses/allProjects.json';
import signedUrl from './responses/signedUrl.json';
import syncProjects from './responses/syncProjects.json';
import deleteToken from './responses/deleteToken.json';
import status from './responses/status.json';
import allEvents_studyCreation from './responses/allEvents_studyCreation.json';
import allEvents_studyCreation_err from './responses/allEvents_studyCreation_err.json';
import allEvents_studyCreation_ing from './responses/allEvents_studyCreation_ing.json';

export const mocks = [
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    error: new Error('something went wrong'),
  },
  {
    request: {
      query: DELETE_FILE,
      variables: {
        kfId: 'SF_Y07IN1HO',
      },
    },
    result: deleteFile,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_5ZPEM167',
      },
    },
    result: fileByKfId,
  },
  {
    request: {
      query: UPDATE_FILE,
      variables: {
        kfId: 'SF_5ZPEM167',
        name: 'foo bar file',
        description: 'Some description here',
        fileType: 'CLN',
      },
    },
    result: updateFile,
  },
  {
    request: {
      query: UPDATE_VERSION,
      variables: {
        versionId: 'FV_DMTKEFQW',
        description: 'version description',
        state: 'APP',
      },
    },
    result: updateVersion,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {
          lastModified: 1567797130510,
          name: 'organize.tsv',
          size: 7,
          type: 'text/tab-separated-values',
          webkitRelativePath: '',
        },
        fileId: 'SF_5ZPEM167',
        description: 'version description',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    result: myProfile,
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    result: devTokens,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    result: createDevToken,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('something went wrong with your event logs request'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    result: allUsers,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    error: new Error('Failed to get user profile information'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    error: new Error('Failed to get the Cavatica project data for this study'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: null,
        deleted: null,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    result: allProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    result: syncProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    error: new Error('Failed to sync Cavatica projects'),
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    error: new Error('Failed to create the new study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_refetch,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {},
        fileId: 'SF_5ZPEM167',
        description: 'my changes',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: FILE_DOWNLOAD_URL,
      variables: {
        studyId: 'SD_00000000',
        fileId: 'SF_5ZPEM167',
        versionId: 'FV_TT8NF09R',
      },
    },
    result: signedUrl,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    error: new Error('Failed to fetch projects information'),
  },
  {
    request: {
      query: DELETE_DEV_TOKEN,
      variables: {
        name: 'my token 1',
      },
    },
    result: deleteToken,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    error: new Error('Failed to create new token'),
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    error: new Error('Failed to fetch tokens information'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_err,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_ing,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: STATUS,
    },
    result: status,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_event_err,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_no_projects,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {orderBy: '-created_at', first: 20},
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    error: new Error('Failed to fetch study information'),
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies_empty,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_INVALID',
      },
    },
    result: studyByKfId_empty,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_INVALIDS',
      },
    },
    result: fileByKfId_empty,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_INVALID',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          externalId: 'benchmark-extensible-e-business',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
        workflows: [],
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: [],
        input: {
          externalId: 'benchmark-extensible-e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          collaborators: [],
        },
      },
    },
    error: new Error('Failed to create the new research study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: 'Test short name input on update',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    error: new Error('Failed to fetch users information'),
  },
];
