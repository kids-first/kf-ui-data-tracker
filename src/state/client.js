import {ApolloClient, HttpLink} from '@apollo/client';
import {createUploadLink} from 'apollo-upload-client';
import {InMemoryCache} from '@apollo/client/cache';
import {KF_STUDY_API, KF_COORD_API} from '../common/globals';

const link = new HttpLink.split(
  operation => operation.getContext().clientName === 'coordinator',
  new HttpLink({uri: `${KF_COORD_API}/graphql`}),
  new createUploadLink({uri: `${KF_STUDY_API}/graphql`}),
);

const policies = {
  typePolicies: {
    StudyNode: {
      fields: {
        kfId: {
          merge(existing, incoming) {
            console.log('merging');
            // Better, but not quite correct.
            return {...existing, ...incoming};
          },
        },
      },
    },
  },
};

export const client = new ApolloClient({
  cache: new InMemoryCache(),
  headers: {
    authorization: localStorage.getItem('accessToken')
      ? `Bearer ${localStorage.getItem('accessToken')}`
      : '',
  },
  link,
});
