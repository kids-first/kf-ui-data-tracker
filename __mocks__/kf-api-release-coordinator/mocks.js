import {GET_STUDY_RELEASES, GET_RELEASED_STUDY} from '../../src/state/queries';
import getStudyReleases from './responses/getStudyReleases.json';
import getStudyReleasesEmpty from './responses/getStudyReleasesEmpty.json';
import allReleaseStudies from './responses/allReleaseStudies.json';

export const coordMocks = {
  getStudyReleases: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleases,
  },
  getStudyReleasesError: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    error: new Error('Failed to fetch releases information'),
  },
  getStudyReleasesEmpty: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleasesEmpty,
  },
  allReleaseStudies: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    result: allReleaseStudies,
  },
  allReleaseStudiesError: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    error: new Error('Failed to fetch study releases information'),
  },
};
