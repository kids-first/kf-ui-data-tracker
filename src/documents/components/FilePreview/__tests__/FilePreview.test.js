import React from 'react';
import wait from 'waait';
import {render, cleanup, fireEvent, act} from '@testing-library/react';
import {MockedProvider} from '@apollo/client/testing';
import {mocks} from '../../../../../__mocks__/kf-api-study-creator/mocks';
import studyByKfId from '../../../../../__mocks__/kf-api-study-creator/responses/studyByKfId';
import {FilePreview} from '../';

afterEach(cleanup);

const mockResponse = (status = 200, returnBody) => {
  global.Headers = jest.fn();
  global.fetch = jest.fn().mockImplementationOnce(() => {
    return new Promise((resolve, reject) => {
      resolve({
        ok: true,
        status,
        text: () => {
          return returnBody ? returnBody : {};
        },
      });
    });
  });
};

it('File Preview renders correctly', async () => {
  mockResponse(200, 'a,b,c,\n1,2,3,\n4,5,6,\n');

  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MockedProvider mocks={mocks}>
      <FilePreview file={file} />
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  act(() => {
    fireEvent.click(tree.getByText('Preview File'));
  });
  await wait();

  expect(tree.container).toMatchSnapshot();
});

it('File Preview renders error correctly', async () => {
  mockResponse(200, 'not a csv or tsv string');

  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MockedProvider mocks={mocks}>
      <FilePreview file={file} />
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  act(() => {
    fireEvent.click(tree.getByText('Preview File'));
  });
  await wait();

  expect(tree.container).toMatchSnapshot();
});
