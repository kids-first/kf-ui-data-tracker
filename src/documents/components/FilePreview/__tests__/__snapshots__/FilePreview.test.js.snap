// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Preview renders correctly 1`] = `
<div>
  <div
    class="ui basic vertical segment ui container"
  >
    <h4
      class="ui grey header"
    >
      File Preview
    </h4>
    <div
      class="ui basic segment scroll-container"
    >
      <div
        class="ui center aligned container"
      >
        <button
          class="ui basic icon primary left labeled button"
        >
          <i
            aria-hidden="true"
            class="eye icon"
          />
          Preview File
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`File Preview renders correctly 2`] = `
<div>
  <div
    class="ui basic vertical segment ui container"
  >
    <h4
      class="ui grey header"
    >
      File Preview
    </h4>
    <div
      class="ui basic segment scroll-container"
    >
      <span
        class="data-grid-container"
        tabindex="0"
      >
        <table
          class="ui definition single line very basic very compact table"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                a
              </th>
              <th
                class=""
              >
                b
              </th>
              <th
                class=""
              >
                c
              </th>
              <th
                class=""
              />
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  1
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  2
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  3
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                />
              </td>
            </tr>
            <tr>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  4
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  5
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                >
                  6
                </span>
              </td>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                />
              </td>
            </tr>
            <tr>
              <td
                class="cell"
              >
                <span
                  class="value-viewer"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </span>
    </div>
  </div>
</div>
`;

exports[`File Preview renders error correctly 1`] = `
<div>
  <div
    class="ui basic vertical segment ui container"
  >
    <h4
      class="ui grey header"
    >
      File Preview
    </h4>
    <div
      class="ui basic segment scroll-container"
    >
      <div
        class="ui center aligned container"
      >
        <button
          class="ui basic icon primary left labeled button"
        >
          <i
            aria-hidden="true"
            class="eye icon"
          />
          Preview File
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`File Preview renders error correctly 2`] = `
<div>
  <div
    class="ui basic vertical segment ui container"
  >
    <h4
      class="ui grey header"
    >
      File Preview
    </h4>
    <div
      class="ui basic segment scroll-container"
    >
      <div
        class="ui icon warning message"
      >
        <i
          aria-hidden="true"
          class="warning sign icon"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            Error parsing file
          </div>
          <p>
            The file was not able to be read to provide a formatted preview. File contents are displayed in raw format below for the first 2048 characters.
          </p>
        </div>
      </div>
      <pre>
        not a csv or tsv string
      </pre>
    </div>
  </div>
</div>
`;
